app/src/main/java/com/dessalines/thumbkey/IMEService.kt

```diff
@@ -68,6 +72,34 @@ class IMEService :
         handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
     }
 
+    // Cursor update Methods
+    override fun onUpdateCursorAnchorInfo(cursorAnchorInfo: CursorAnchorInfo) {
+        super.onUpdateCursorAnchorInfo(cursorAnchorInfo)
+
+        // If the cursor has moved at all vertically, or more than a small amount horizontally, the cursor has changed and multitap should be blocked.
+        // The horizontal buffer is because the cursor moves slightly based on the size of some of the characters (i.e '?') moving the cursor a little bit.
+        // It would be better to not use a magic number of 15, but I don't know what the ideal buffer is and it seems to work well, even when moving the cursor right before the multitap character
+        if (insertionMarkerBaseline != cursorAnchorInfo.getInsertionMarkerBaseline() ||
+            abs(cursorAnchorInfo.getInsertionMarkerHorizontal() - insertionMarkerHorizontal) > 15f
```

Comment:
I spose this should be something like 2 or 3? The multitaps should only use 1-2 characters max anyway. Needs testing  to be sure.

Reply:
I don't think the amount refers to characters, when I only did 1 or 2 the bug was always present, but with 15 the bug is not present even 1 character away (e.g. typing "the", double-space, tap between h and e, space gives "th e, " rather than "t. e, ")

Reply:
Setting it too high allows the bug to show up if you tap near where you double tapped, setting it too low causes multitap to stop working, especially once you hit '?' in the cycle, since '?' is wide enough to move the cursor more than characters like '.' do

Reply:
That said, there probably is a more optimal value